### PR TITLE
Fix #761 - Update retirement manager to use identity keys

### LINKED ISSUE #761: RetirementManager should use identity for keys
currently, retirement manager tracks retirement based on EntityMessage which has no hashCode or equals contract. Use a wrapper object to use object identity for keys.

### PR DESCRIPTION
**Pull request by myronkscott**

### CODE REVIEW

File: dso-l2/src/main/java/com/tc/objectserver/handler/RetirementManager.java [hunk 1]
```diff
@@ -49,59 +49,57 @@
  * message, per key.
  */
 public class RetirementManager {
-  private final Map<EntityMessage, LogicalSequence> currentlyRunning;
-  private final Map<EntityMessage, LogicalSequence> waitingForDeferredRegistration;
+  private final Map<Identity<EntityMessage>, LogicalSequence> currentlyRunning;
```
Comment (ramsai1729) [inline dso-l2/src/main/java/com/tc/objectserver/handler/RetirementManager.java]: how about using https://docs.oracle.com/javase/8/docs/api/java/util/IdentityHashMap.html?

File: dso-l2/src/main/java/com/tc/objectserver/handler/RetirementManager.java [hunk 2]
```diff
@@ -163,8 +157,9 @@ private void traverseDependencyGraph(List<Retiree> toRetire, LogicalSequence com
     }
   }
 
-  public synchronized void deferRetirement(EntityMessage invokeMessageToDefer, EntityMessage laterMessage) {
+  public void deferRetirement(EntityMessage invokeMessageToDefer, EntityMessage laterMessage) {
```
Comment (ramsai1729) [inline dso-l2/src/main/java/com/tc/objectserver/handler/RetirementManager.java]: could you explain why synchronized is not needed here?

File: dso-l2/src/main/java/com/tc/objectserver/handler/RetirementManager.java [hunk 3]
```diff
@@ -190,16 +185,16 @@ public synchronized void entityWasDestroyed() {
     Assert.assertTrue(this.waitingForDeferredRegistration.isEmpty());
   }
   
-  public Map<String, Object> getState() {
+  public synchronized Map<String, Object> getState() {
```
Comment (ramsai1729) [inline dso-l2/src/main/java/com/tc/objectserver/handler/RetirementManager.java]: doesn't this block request processor threads?

File: dso-l2/src/main/java/com/tc/objectserver/handler/RetirementManager.java (hunk 2 above)
Comment (myronkscott) [reply dso-l2/src/main/java/com/tc/objectserver/handler/RetirementManager.java]: that is a mistake will fix.

File: dso-l2/src/main/java/com/tc/objectserver/handler/RetirementManager.java (hunk 3 above)
Comment (myronkscott) [reply dso-l2/src/main/java/com/tc/objectserver/handler/RetirementManager.java]: yes but the alternative is a random CME.  We should look into refactoring this class to be more concurrent at some point.

**Pull request merged**